go-controller/pkg/cni/ovs.go

```diff
@@ -170,6 +171,12 @@ func waitForPodFlows(mac string, ifAddrs []*net.IPNet, ifaceName, ifaceID string
 				return false, nil
 			}
 		}
+
+		// check for cancelation
+		if atomic.LoadUint32(cancel) > 0 {
```

Comment:
we should use a `context` and wait in parallel, this still waits for the ovs operations that may hang or take time

Reply:
could we wrap the entire cmdAdd in the context and cancel any part of it instead of just this function?

Reply:
yeah, like having a global map with the ctxs, so we just cancel from the top and it cascade the cancel down

Reply:
Done.